subsys/net/lib/conn_mgr/conn_mgr.c

```diff
@@ -24,6 +24,16 @@ LOG_MODULE_REGISTER(conn_mgr, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
 
 uint16_t iface_states[CONN_MGR_IFACE_MAX];
 
+/**
```

Comment:
I have an impression that those `conn_mgr` changes are not tightly bound to the generic connectivity framework introduction. Perhaps they could be provided in a separate commit to simplify the review process?

Reply:
And perhaps come with a modification of a sample to demonstrate the new approach?